pkg/replicatedfile/cache.go

```diff
@@ -0,0 +1,119 @@
+package replicatedfile
+
+import (
+	"encoding/json"
+	"fmt"
+	"os"
+	"path"
+	"time"
+
+	"github.com/usrbinapp/usrbin-go/pkg/updatechecker"
+)
+
+const (
+	cacheTTLInHours = 4
+
+	replicatedDir = ".replicated"
+	cacheFile     = "cache.json"
+)
+
+type cache struct {
+	CachedVersion     string                    `json:"cachedVersion,omitempty"`
+	UpdateCheckerInfo *updatechecker.UpdateInfo `json:"updateCheckerInfo,omitempty"`
+}
+
+var (
+	Cache *cache
+)
+
+func init() {
+	var err error
+	Cache, err = loadCache()
+	if err != nil {
+		fmt.Println(err)
```

Comment:
can you add some contectual information in this error? something like

```suggestion
		fmt.Printf("Failed to load cache file: %v\n", err)
```